### PR TITLE
Azure pipelines

### LINKED ISSUE #656: Request: Build, Test and Release tslint-microsoft-contrib using Azure Pipelines
This is a request to implement continuous integration builds that test tslint-microsoft-contrib using Azure Pipelines.  The pipeline would be developed to run tests across various versions of node on Windows and Linux on every commit.  A release pipeline could also be used for releasing/publishing to npm and NuGet and automate the versioning described in https://github.com/Microsoft/tslint-microsoft-contrib/wiki/Releases.   

Note: This request is to only build and release using Azure Pipelines and IS NOT a suggestion to move the GitHub repo to Azure Repos.  

Note that I have already reserved the account  "tslint-microsoft-contrib" on Azure Repos and added some of the community members as admins.  https://dev.azure.com/tslint-microsoft-contrib/tslint-microsoft-contrib

### PR DESCRIPTION
**Pull request by apawast**
#### PR checklist

-   [x] Addresses an existing issue: fixes #656 , refer to PR #655 
-   [x] New feature, bugfix, or enhancement
    -   [ ] Includes tests
-   [ ] Documentation update

#### Overview of change:
Add support for CI using Azure Pipelines

#### Is there anything you'd like reviewers to focus on?
I have created a build pipeline for tslint-microsoft-contrib project here: https://dev.azure.com/ms/tslint-microsoft-contrib
Microsoft employees should all have access to this already. Currently it's building off of my fork of the project, however, you or Will Smythe can change it to point to your repo.

I had to add the reporter to publish the results:

tests

However, I found a new solution so that the reporter will not affect developers working on this project.

### CODE REVIEW

**Comment by apawast** [comment]
@JoshuaKGoldberg

File: package.json [hunk 1]
```diff
@@ -83,12 +86,14 @@
     "chalk": "^2.4.1",
     "chokidar": "^2.0.4",
     "cpy-cli": "^2.0.0",
+    "cross-var": "^1.1.0",
```
Comment (JoshuaKGoldberg) [inline package.json]: Heh, TIL about this. Thanks! Seems useful 😊

**Comment by JoshuaKGoldberg** [comment]
Whoops, I accidentally pushed a commit to your `azure-pipelines` branch. Sorry about that 😄 force-pushed it out of the history.

**Comment by apawast** [comment]
> Well this is spooky. Here are three PRs with failing tests:
> 
> * [apawast#1](https://github.com/apawast/tslint-microsoft-contrib/pull/1): failing .ts.lint test; correctly reports the failure
> * [apawast#2](https://github.com/apawast/tslint-microsoft-contrib/pull/2): failing Mocha test with the `cross-var` changes
> * [apawast#3](https://github.com/apawast/tslint-microsoft-contrib/pull/3): failing Mocha test without any package changes
> 
> The .ts.lint test correctly fails, but neither build with failing Mocha tests picked up on it. That seems like a blocker _(unless there was user error on my end?)_.

I reproduced those test failure locally and realized that the message that you changed isn't what it's testing for, it's testing for true or false. The message could be anything and it'll pass regardless. I changed the tests to say false instead of true and it failed locally immediately. I didn't want to have random failing builds in the pipeline setup so I didn't push but if you want I can reproduce it in a different branch or something.

**Comment by JoshuaKGoldberg** [comment]
Ha, user error on my end indeed. I'll take another look. Thanks!

**Comment by JoshuaKGoldberg** [comment]
@apawast  https://github.com/apawast/tslint-microsoft-contrib/pull/3 is correctly on failed unit tests after reverting changes to `package.json`. That seems to indicate changes to `package.json` in this PR can be reverted, right? Unless I'm (again) missing something, could you please?

Also, in case it's useful, some feedback about Azure Pipelines itself:

* Going directly from the links in https://github.com/apawast/tslint-microsoft-contrib/pull/3, it takes several clicks through pages to get to the actual error log. That's a little annoying as a user; I just want to directly go to the error description.
* The log in https://dev.azure.com/apawast/tslint/_build/results?buildId=212 does not have colored text the way Circle and Travis. It's very convenient to be able to quickly scan through logs for the red error text (and inconvenient not to be able to).

I'm happy to send you / your team more detailed feedback here or via another channel if you'd like.

**Comment by apawast** [comment]
The changes I made to `package.json` are for test reporting. Azure Pipelines also has rich test reporting which works well with commonly used test runners including the mocha-junit-reporter. If you switch from the Logs tab to the Tests tab of a build you can see the test reporting. The default setting of the pipeline is to show failed/aborted tests so if a test failed, you wouldn't have to look through the logs, you could just view them in the Test tab first thing (which helps with the colorless logs issue). However, I have also added a force color variable to ensure the logs show colored text too.

For the issue about getting through several clicks, that is just how Github handles getting to the pipeline from the repo. We can't do anything about it, however, I have heard there is a project in the works to fix this.

Any other feedback you have, you can email it to me at apwast@microsoft.com and I will be happy to pass it on to the appropriate people!

**Comment by JoshuaKGoldberg** [comment]
> The changes I made to package.json are for test reporting. Azure Pipelines also has rich test reporting which works well with commonly used test runners including the mocha-junit-reporter

Yes, but tslint and tslint-microsoft-contrib don't use a commonly used test runner. From #655:

> Also worth to add: there are new npm tasks that should generate report from Mocha. However #489 will migrate almost all test away from Mocha, and only test for few utils and deprecated formatters will left as mocha tests. I do not think that it worth it to have report just for them.

For this PR to be merged, there should be no `package.json` or `package-lock.json` changes.

> I have also added a force color variable

Awesome, thanks!

> that is just how Github handles getting to the pipeline from the repo. We can't do anything about it

No, it's how _you_ handle getting to the pipeline from the repo 😄. See these PR on CircleCI: https://github.com/general-language-syntax/GLS/pull/641. Clicking [Details](https://circleci.com/gh/general-language-syntax/GLS/922?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) takes you directly to a well-formatted display of which tasks failed.

Anyway, this is bikeshedding - the real sticking point here is the package.json changeset.

**Comment by apawast** [comment]
Okay I see. I have no problems removing the changes from them if you're okay with no test reporting (at least the logs will be colorful and I'm sure that's what you're used to looking at them anyways).

I don't really know much about that repo to pipelines process, I was just relaying what other PMs informed me. But I'll definitely look into it!

**Comment by apawast** [comment]
@JoshuaKGoldberg I changed them to match your master. Everything look okay?

**Comment by JoshuaKGoldberg** [comment]
Node 6 build failures are unrelated.

**Comment by JoshuaKGoldberg** [comment]
Thanks for setting this up @apawast! We'll try this out for a while and send you feedback as we come across it.

**Comment by JoshuaKGoldberg** [comment]
Oh, just a heads up, I don't have permissions to add it as an app to this repository. You'll want to work with @HamletDRC for that, I believe. It might need an admin who's a member of the @microsoft organization.

**Pull request merged**